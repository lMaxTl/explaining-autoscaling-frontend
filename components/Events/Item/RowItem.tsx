import { Grid, ThemeIcon, Text } from "@mantine/core";
import { IconArrowRightBar, IconArrowsMaximize, IconArrowsMinimize, IconCheck, IconExclamationMark } from "@tabler/icons";
import Link from "next/link";
import { formatDate } from "../../../helpers/helper";
import { RowItemProps } from "./RowItem.dto";
import { useStyles } from "./RowItem.styles";

export default function RowItem({index, element}: RowItemProps) {
    const { classes, cx } = useStyles();
    const link = `/events/${element.name}-event-${element._id}`;
    return (
        <Link key={index} href={link} legacyBehavior>
            <tr key={element._id} className={cx(classes.linkHoverEffect)}>
                <td>{formatDate(element.createdAt)}</td>
                <td>{element.namespace}/{element.name}</td>
                <td>
                    <Grid align="center">
                        <Grid.Col span={3}>
                            {element.scalingType == 'scaleOut' &&
                                (<IconArrowsMaximize className={cx(classes.distanceToTop)} />)}

                            {element.scalingType == 'scaleIn' &&
                                (<IconArrowsMinimize className={cx(classes.distanceToTop)} />)}
                        </Grid.Col>
                        <Grid.Col span={9}>
                            {element.scalingType == 'scaleOut' &&
                                (<Text c="green"> {element.scalingType} </Text>)}

                            {element.scalingType == 'scaleIn' &&
                                (<Text c="yellow"> {element.scalingType} </Text>)}
                        </Grid.Col>
                    </Grid>
                </td>
                <td>{element.metricType}</td>
                <td>
                    <Grid align="center">
                        <Grid.Col span={1}>{element.oldReplicaSize}</Grid.Col>
                        <Grid.Col span={4}><IconArrowRightBar className={cx(classes.distanceToTop)} /></Grid.Col>
                        <Grid.Col span={1}>{element.replicaSize}</Grid.Col>
                    </Grid>
                </td>
                <td>
                    <Grid align="center">
                        <Grid.Col span={3}>

                            {element.reason == 'SuccessfulRescale' &&
                                (<ThemeIcon radius="xl" color="green" className={cx(classes.distanceToTop)}><IconCheck /></ThemeIcon>)}

                            {element.reason == 'UnableToScale' &&
                                (<ThemeIcon radius="xl" color="red" className={cx(classes.distanceToTop)}><IconExclamationMark /></ThemeIcon>)}
                        </Grid.Col>
                        <Grid.Col span={9}>{element.reason}</Grid.Col>
                    </Grid>
                </td>
            </tr>
        </Link>
    );
}