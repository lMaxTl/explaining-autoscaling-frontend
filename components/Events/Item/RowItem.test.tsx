import React from 'react';
import { render } from '@testing-library/react';
import RowItem from './RowItem';

describe('RowItem', () => {
    it('should render the component', () => {
        const element = {
            _id: '123',
            name: 'test-deployment',
            namespace: 'test',
            createdAt: '2022-01-01',
            message: 'Test message',
            reason: 'SuccessfulRescale',
            replicaSize: 2,
            oldReplicaSize: 1,
            scalingType: 'scaleOut',
            metricType: 'CPU',
            __v: 0,
        };
        const { container } = render(<RowItem index={0} element={element} />);

        expect(container.firstChild).toMatchSnapshot();
    });
});
