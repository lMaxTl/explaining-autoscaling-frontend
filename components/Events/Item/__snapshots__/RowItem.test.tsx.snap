// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowItem should render the component 1`] = `
<tr
  class="mantine-60yz8o"
>
  <td>
    2022-1-1 1:00:0
  </td>
  <td>
    test
    /
    test-deployment
  </td>
  <td>
    <div
      class="mantine-Grid-root mantine-1azqbd9"
    >
      <div
        class="mantine-Col-root mantine-pzrbb4"
      >
        <svg
          class="mantine-5c15xa"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <polyline
            points="16 4 20 4 20 8"
          />
          <line
            x1="14"
            x2="20"
            y1="10"
            y2="4"
          />
          <polyline
            points="8 20 4 20 4 16"
          />
          <line
            x1="4"
            x2="10"
            y1="20"
            y2="14"
          />
          <polyline
            points="16 20 20 20 20 16"
          />
          <line
            x1="14"
            x2="20"
            y1="14"
            y2="20"
          />
          <polyline
            points="8 4 4 4 4 8"
          />
          <line
            x1="4"
            x2="10"
            y1="4"
            y2="10"
          />
        </svg>
      </div>
      <div
        class="mantine-Col-root mantine-1b3zvi6"
      >
        <div
          class="mantine-Text-root mantine-dp00i"
        >
           
          scaleOut
           
        </div>
      </div>
    </div>
  </td>
  <td>
    CPU
  </td>
  <td>
    <div
      class="mantine-Grid-root mantine-1azqbd9"
    >
      <div
        class="mantine-Col-root mantine-pzu1jq"
      >
        1
      </div>
      <div
        class="mantine-Col-root mantine-svfr8w"
      >
        <svg
          class="mantine-5c15xa"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <path
            d="M18 15l3 -3l-3 -3"
          />
          <path
            d="M3 12h18"
          />
          <path
            d="M3 9v6"
          />
        </svg>
      </div>
      <div
        class="mantine-Col-root mantine-pzu1jq"
      >
        2
      </div>
    </div>
  </td>
  <td>
    <div
      class="mantine-Grid-root mantine-1azqbd9"
    >
      <div
        class="mantine-Col-root mantine-pzrbb4"
      >
        <div
          class="mantine-ThemeIcon-root mantine-j03v3r"
        >
          <svg
            class="icon icon-tabler icon-tabler-check"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
              stroke="none"
            />
            <path
              d="M5 12l5 5l10 -10"
            />
          </svg>
        </div>
      </div>
      <div
        class="mantine-Col-root mantine-1b3zvi6"
      >
        SuccessfulRescale
      </div>
    </div>
  </td>
</tr>
`;
